From 7764aa273054bf351c834f5670b51407331b009f Mon Sep 17 00:00:00 2001
From: "Suresh, Prashanth" <prashanth.suresh@intel.com>
Date: Wed, 15 Jan 2025 22:29:08 +0530
Subject: [PATCH] Fix compilation failure for QEMU 9.1

---
 include/ui/dmabuf.h | 20 +++++++++++++++++++-
 ui/dmabuf.c         | 19 -------------------
 2 files changed, 19 insertions(+), 20 deletions(-)

diff --git a/include/ui/dmabuf.h b/include/ui/dmabuf.h
index 07e8178d7..3e217e53b 100644
--- a/include/ui/dmabuf.h
+++ b/include/ui/dmabuf.h
@@ -10,7 +10,25 @@
 #ifndef DMABUF_H
 #define DMABUF_H
 
-typedef struct QemuDmaBuf QemuDmaBuf;
+typedef struct QemuDmaBuf {
+    int       fd;
+    uint32_t  width;
+    uint32_t  height;
+    uint32_t  stride;
+    uint32_t  fourcc;
+    uint64_t  modifier;
+    uint32_t  texture;
+    uint32_t  x;
+    uint32_t  y;
+    uint32_t  backing_width;
+    uint32_t  backing_height;
+    bool      y0_top;
+    int       fence_fd;
+    bool      allow_fences;
+    bool      render_sync;
+    bool      draw_submitted;
+} QemuDmaBuf;
+
 
 QemuDmaBuf *qemu_dmabuf_new(uint32_t width, uint32_t height,
                             uint32_t stride, uint32_t x,
diff --git a/ui/dmabuf.c b/ui/dmabuf.c
index 0adaea008..d8ed6731d 100644
--- a/ui/dmabuf.c
+++ b/ui/dmabuf.c
@@ -10,25 +10,6 @@
 #include "qemu/osdep.h"
 #include "ui/dmabuf.h"
 
-struct QemuDmaBuf {
-    int       fd;
-    uint32_t  width;
-    uint32_t  height;
-    uint32_t  stride;
-    uint32_t  fourcc;
-    uint64_t  modifier;
-    uint32_t  texture;
-    uint32_t  x;
-    uint32_t  y;
-    uint32_t  backing_width;
-    uint32_t  backing_height;
-    bool      y0_top;
-    int       fence_fd;
-    bool      allow_fences;
-    bool      render_sync;
-    bool      draw_submitted;
-};
-
 QemuDmaBuf *qemu_dmabuf_new(uint32_t width, uint32_t height,
                             uint32_t stride, uint32_t x,
                             uint32_t y, uint32_t backing_width,
-- 
2.43.0

